### ISSUE TITLE
Example of angular code

### ISSUE THREAD

**Issue by lambert-velir** [javoire/browserify-ng-html2js#11]
Hi-

I'm trying to pipe my index.js file though this transform to add my templates to the $templateCache.  Is this transform supposed to parse out any templateUrl declarations?  In my index.js, I have: 

```javascript
// index.js
app.config(function($routeProvider){
    $routeProvider
        .when("/bar", {
            templateUrl: "bar/bar.html", 
            controller: require("./bar/barCtrl.js")
        });
});
```

My gulp task looks like this: 

```javascript
// gulp task
browserify("index.js")
    .transform(ngHtml2Js({
      module: "templates"
    }))
    .bundle()
    .pipe(source("bundle-index.js"))
    .pipe(gulp.dest("/js-generated/"));
```

Is this transform supposed to recognize bar/bar.html and add it to my bundle?

**Comment by mchambaud** [comment]
@lambert-velir try requiring your template.

`
// index.js
require('you-template.html');

app.config(function($routeProvider){
    $routeProvider
        .when("/bar", {
            templateUrl: "bar/bar.html", 
            controller: require("./bar/barCtrl.js")
        });
});
`

**Comment by lambert-velir** [comment]
Cool, ``` require("./bar/bar.html");``` works, thanks!

@javoire, you might want to consider adding this to the documentation.  This wasn't apparent to me.

**Comment by javoire** [comment]
Hmm yeah. I'm thinking though that the very concept behind browserify is to `require()` the files you need in your final js output, i.e. all the files that will be sent and processed through browserfys pipeline. So if a html needs to be processed by browserify and compiled to an angular template it needs to be required :) But maybe it won't hurt to show that in an example. I'll give it some thoughts. Cheers

**Comment by mchambaud** [comment]
@javoire it would be nice if we did not need to require the templates!

**Comment by javoire** [comment]
True. But that's how browserify works! You require _all_ the source files you want to be a part of your bundle. How would browserify know where your templates are if you don't require them? ^^

**Comment by mchambaud** [comment]
It would be quite interesting if a node script could look for templateUrl: "path/to/template.html" and inject a require tag before running browserify. 

Hopefully it wouldn't be too costly to run.

**Comment by javoire** [comment]
Yeah I guess that could make sense ^^

**Comment by carlosacabrera** [comment]
Hey guys, maybe I'm doing something but I'm using the require and it is working. However, my karma testing are failing with  SyntaxError: Unexpected token (1:0) while parsing file:...

I already installed karma-ng-html2js-preprocessor and followed its documentation. The issue is that Karma is getting that require('file.html') and failing. Any hints?

**Comment by javoire** [comment]
Hmm, well karma would have to use browserify somehow also, is it?

**Comment by javoire** [comment]
What you can do otherwise, is to bundle (browserify) the code before running karma, so that karma will take the finished `bundle.js` (or whatever you've named you output file) is it's input in `karma.conf.json`

**Comment by carlosacabrera** [comment]
Hi Jonatan, to bundle the file previous to running Karma is not an option due to how the application and the tests have been set up in the past. We are using browserify as a preprocessor in Karma. I'm currently investigation how can I add ng-html2js as part of the pre-processor.

**Comment by toru-sean** [comment]
i agree that it would be better if the documentation said that you need to require the templates in your code. I only started using browserify today so it wasn't at all obvious to me. Great plugin though!

**Comment by javoire** [comment]
agreed :) re-opening

**Comment by andrewdonskikh** [comment]
Please, man! Add this to the documentation - I spent 2 hours until got this.

**Comment by javoire** [comment]
closed in #37
